lib/streamlit/config.py

```diff
@@ -966,18 +966,18 @@ def _set_development_mode():
     development.is_development_mode = get_option("global.developmentMode")
 
 
-def on_config_parsed(func):
+def on_config_parsed(func, force_connect=False):
     """Wait for the config file to be parsed then call func.
 
     If the config file has already been parsed, just calls fun immediately.
 
     """
-    if _config_file_has_been_parsed:
-        func()
-    else:
+    if force_connect or not _config_file_has_been_parsed:
         # weak=False, because we're using an anonymous lambda that
         # goes out of scope immediately.
         _on_config_parsed.connect(lambda _: func(), weak=False)
+    else:
+        func()
```

Comment:
I think there's something better we need to do with the config system, which requires more thought.

But in the meantime, can we refactor this current solution for readability? 

For example, we could have two signals:

```py
if file has not been parsed:
  _on_config_parsed.connect(lambda _: func(), weak=False)

if command-line args have not been parsed:
  _on_args_parsed.connect(lambda _: func(), weak=False)
```